sdk/n8nsdk/api/openapi-source/v1/handlers/credentials/credentials.service.ts

```
@@ -91,11 +149,111 @@ export async function saveCredential(
+		const credentialsService = Container.get(CredentialsService);
+
+		// Decrypt existing data to access oauthTokenData
+		const decryptedData = credentialsService.decrypt(existingCredential as CredentialsEntity, true);
+
+		validateExternalSecretsPermissions(user, updateData.data, decryptedData);
+
+		if (isProjectScopedExternalSecretsEnabled() && decryptedData) {
+			// eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain -- credential will always have an owner
+			const projectOwningCredential = existingCredential.shared?.find(
+				(shared) => shared.role === 'credential:owner',
+			)!;
```

Biome (2.4.4)

[error] Forbidden non-null assertion after optional chaining.

(lint/suspicious/noNonNullAssertedOptionalChain)